ci/bin/lint.dart

```diff
@@ -0,0 +1,241 @@
+  final ProcessRunnerResult masterResult = await processRunner
+      .runProcess(<String>['git', 'diff', '--name-only', '--diff-filter=ACMRT', mergeBase]);
+  result.addAll(utf8.decode(masterResult.stdout).split('\n').where(isNonEmptyString));
+  return result.map<File>((String filePath) => File(path.join(repoPath.path, filePath))).toList();
+}
+
+Future<List<File>> dirContents(Directory dir) {
+  final List<File> files = <File>[];
+  final Completer<List<File>> completer = Completer<List<File>>();
+  final Stream<FileSystemEntity> lister = dir.list(recursive: true);
+  lister.listen((FileSystemEntity file) => file is File ? files.add(file) : null,
+      onError: (Object e) => completer.completeError(e), onDone: () => completer.complete(files));
+  return completer.future;
+}
+
+File buildFileAsRepoFile(String buildFile, Directory repoPath) {
+  // Removes the "../../flutter" from the build files to make it relative to the flutter
+  // dir.
+  final String relativeBuildFile = path.joinAll(path.split(buildFile).sublist(3));
+  final File result = File(path.join(repoPath.absolute.path, relativeBuildFile));
+  print('Build file: $buildFile => ${result.path}');
+  return result;
+}
+
+Future<bool> shouldIgnoreFile(File file) async {
+  if (path.split(file.path).contains('third_party')) {
+    return true;
+  } else {
+    final RegExp exp = RegExp(r'//.*FLUTTER_NOLINT');
+    await for (String line
+        in file.openRead().transform(utf8.decoder).transform(const LineSplitter())) {
+      if (exp.hasMatch(line)) {
+        return true;
+      } else if (line.isNotEmpty && line[0] != '\n' && line[0] != '/') {
+        // Quick out once we find a line that isn't empty or a comment.  The
+        // FLUTTER_NOLINT must show up before the first real code.
+        return false;
+      }
+    }
+    return false;
+  }
+}
+
+void _usage(ArgParser parser) {
+  stderr.writeln('lint.dart [--help] [--lint-all] [--verbose] [--diff-branch]');
+  stderr.writeln(parser.usage);
+  exit(0);
+}
+
+bool verbose = false;
+
+void main(List<String> arguments) async {
+  final ArgParser parser = ArgParser();
+  parser.addFlag('help', help: 'Print help.');
+  parser.addFlag('lint-all',
+      help: 'lint all of the sources, regardless of FLUTTER_NOLINT.', defaultsTo: false);
+  parser.addFlag('verbose', help: 'Print verbose output.', defaultsTo: verbose);
+  parser.addOption('repo', help: 'Use the given path as the repo path');
+  parser.addOption('compile-commands',
+      help: 'Use the given path as the source of compile_commands.json. This '
+          'file is created by running tools/gn');
+  parser.addOption('checks',
+      help: 'Perform the given checks on the code. Defaults to the empty '
+          'string, indicating all checks should be performed.',
+      defaultsTo: '');
+  final ArgResults options = parser.parse(arguments);
+
+  verbose = options['verbose'] as bool;
+
+  if (options['help'] as bool) {
+    _usage(parser);
```

Comment:
Shouldn't execution happen after _usage?

Reply:
Yes, and it does: `_usage` exits.